### PR TITLE
Remove develop_server.sh in favour of pelican serving static files itself - close #2176

### PR DESCRIPTION
**Pull request by Lucas-C**
Competing static site generators integrate the functionality of regenerating
content and serving it into their main executable. In pelican this
functionality used to be in an external script `develop_server.sh` which
resides in the blog base directory. This has the disadvantage that changes in
pelican can break the `develop_server.sh` scripts which will not automatically
be upgraded together with pelican by package managers. Thus, pelican should
integrate this functionality into its main executable.

To this end, this commit removes `develop_server.sh` and adds three command
line options to the pelican executable:

 * `-l/--listen` starts the HTTP server (`-s/--serve` was already taken)
 * `-p/--port` specifies the port to listen at
 * `-b/--bind` specifies the IP to bind to

`--listen` and `--autoreload` can be used together to achieve the same
effect that other static site generators offer: Serve files via HTTP
while at the same time auto-generating the content.

Since the `develop_server.sh` script was removed, pelican-quickstart looses the
`develop` option.

Since the `develop_server.sh` script was removed, the Makefile looses the
`stopserver` target and the `devserver` target is replaced by running `pelican
-l` in the foreground.

Since pelican now offers the `--listen` option, the fabfile uses that instead
of starting the socketserver itself.

### CODE REVIEW

**Comment by Lucas-C** [comment]
Tested locally with my pelican blog

File: pelican/tools/pelican_quickstart.py [hunk 1]
```diff
@@ -263,9 +263,7 @@ def main():
 
     automation = ask('Do you want to generate a Fabfile/Makefile '
                      'to automate generation and publishing?', bool, True)
-    develop = ask('Do you want an auto-reload HTTP script '
-                  'to assist with theme and site development?', bool, True)
-
+
```
Comment (avaris) [inline pelican/tools/pelican_quickstart.py]: Can you clean the whitespace? It's issuing a `flake` error.

**Comment by justinmayer** [comment]
@avaris: It seems the whitespace has been cleaned here. Any other thoughts regarding this PR?

**Comment by avaris** [comment]
I'll take a detailed look this afternoon but there seems to be a merge conflict right now.

**Comment by justinmayer** [comment]
@avaris: Quite right. Sorry I missed that.

@Lucas-C: Would you have a look at the conflict in `pelican/__init__.py` and resolve it accordingly?

**Comment by Lucas-C** [comment]
Merge conflict fixed.

**Comment by justinmayer** [comment]
@Lucas-C: Thanks for resolving the merge conflict.

@avaris: Any thoughts on this PR? Would nice to merge it before releasing 3.8. 😁

**Comment by justinmayer** [comment]
Many thanks to @josch and @Lucas-C for the contribution and to @avaris for reviewing! 🎉

**Pull request merged**